File: Web/src/app/models/access-control/employee.ts
import { Role } from './role';

export class Employee {
  Id: number;
  Email: string;
  Name: string;
  active: boolean = false;
  RoleId: number;
  Role: Role;

constructor(param: any) {
  this.Id = param.Id;
  this.Email = param.Email,
  this.Name = param.Name,
  this.active = this.active;
  this.RoleId = this.RoleId;
  this.Role = param.Role;
}
}

File: Api/Timesheet/Entities/Employee.cs
using Microsoft.EntityFrameworkCore;
using Timesheet.Entities.Accesscontrol;

namespace Timesheet.Entities
{

    public class Employee : EntityBase
    {

        public string Email { get; set; }
        public string Name { get; set; }
        public bool Active { get; set; }
        public int RoleId { get; set; }
        public Role? Role { get; set; }

        public virtual ICollection<EmployeeAttendance>? EmployeeAttendances { get; set; }
        public virtual ICollection<EmployeeScheduleType>? EmployeeScheduleTypes { get; set; }
        public virtual ICollection<TimeOff>? TimesOff { get; set; }
        public virtual ICollection<Vacation>? Vacations { get; set; }

    }
}
